zerver/lib/actions.py

```diff
@@ -2736,6 +2736,29 @@ def validate_message_edit_payload(
         raise JsonableError(_("Widgets cannot be edited."))
 
 
+def can_edit_content_or_topic(
+    message: Message,
+    user_profile: UserProfile,
+    is_no_topic_msg: bool,
+    content: Optional[str] = None,
+    topic_name: Optional[str] = None,
+) -> bool:
+    if message.sender == user_profile:
+        return True
+
+    if content is not None:
+        return False
+
+    if topic_name is not None:
+        if not (
+            is_no_topic_msg
+            or user_profile.is_realm_admin
+            or user_profile.realm.allow_community_topic_editing
+        ):
+            return False
+    return True
```

Comment:
I'd think about reorganizing this a bit more for readability.

Reply:
I thought a bit about it but cannot figure out the way ot reorganize this.
I think we need to reorganize only last conditional and that will be the below mentioned code at the end, and along with the comments, readability would be improved.
```
if not (is_no_topic_msg or user_profile.can_edit_topic_of_any_message()):
            return False
```